Adding support for TimeSpan and DateTimeOffset to ConfigurationBinder

MichaCo opened the pull request:
As discussed
#216

dnfclas commented:
Hi __@MichaCo__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

HaoK commented on src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs:
```diff
@@ -6,6 +6,7 @@
 using System.Linq;
 using System.Reflection;
 using Microsoft.Framework.Configuration.Binder;
+using System.Globalization;
```
nit: sort

HaoK commented on test/Microsoft.Framework.Configuration.Binder.Test/ConfigurationBinderTests.cs:
```diff
@@ -132,6 +133,36 @@ public void ShouldBeIgnoredTests(string property)
+            Assert.Equal(2015, options.DateTimeOffset.Year);
+        }
+
+        [Fact]
```
Might be worth adding a failure/parse issue for both of these to ensure stuff doesn't boom too badly

MichaCo commented:
@HaoK I've added some more unit tests. Now also verifying that all works fine with different cultures.

I also fixed two tests which were failing for me on a German Windows because of the string comparison against a localized exception message.

Let me know what you think, Thanks!

HaoK commented:
:shipit:

MichaCo commented:
@kirthik @HaoK I merged my changes to the latest dev branch for you. 
Could someone merge it? Or let me know if something is wrong. Thanks!

davidfowl commented:
Why did all of the project.json files change?

MichaCo commented:
mh good question, idk, seems git did that during merge

davidfowl commented:
@MichaCo looks like something got messed up with the merge

MichaCo commented:
I will re build the pull request
